Statistics Page

song50119 opened the pull request:
This PR continue process #423

oluwandabira committed:
Statistics page

oluwandabira committed:
Use full names to satisfy structcheck

Commit:
Refactor Statistics Page
- Add spacing to page
- Show network name, bestblock, data directory, wallet data size, number of txs and wallet

Commit:
Remove margin right page container

Commit:
Remove page separator and use from theme

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+func (pg *statPage) lineSeparator() layout.Widget {
+	return func(gtx C) D {
+		return pg.theme.Separator().Layout(gtx)
+	}
```
```suggestion
		return pg.theme.Separator().Layout
```

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+	txs         **wallet.Transactions
+	theme       *decredmaterial.Theme
+	l           layout.List
+	startupTime string
```
Not needed and can be removed.
```suggestion
	
```

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+			Axis: layout.Vertical,
+		},
+	}
+	pg.startupTime = time.Now().String()
```
Not needed and can be removed.
```suggestion

```

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+		pg.lineSeparator(),
+		item("Peers connected", strconv.Itoa(int(pg.syncStatus.ConnectedPeers))),
+		pg.lineSeparator(),
+		item("Uptime", pg.startupTime),
```
```suggestion
		item("Uptime", time.Now().String()),
```

oshorefueled replied:
time.Now() returns the current time, Uptime should be how long the app has been running.

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+	background := pg.common.theme.Color.Surface
+	card := pg.common.theme.Card()
+	card.Color = background
+	inset := layout.Inset{
+		Top:    values.MarginPadding12,
+		Bottom: values.MarginPadding12,
+		Right:  values.MarginPadding16,
+		Left:   values.MarginPadding16,
+	}
```
should the values need be updated at any point, you have have one point of edit.
```suggestion
       mtb := values.MarginPadding12,
       mlr := values.MarginPadding16,
	inset := layout.Inset{
		Top:    mtb,
		Bottom: mtb,
		Right:  mrl,
		Left:   mrl,
	}
```

oshorefueled replied:
I don't think this change is necessary. The reason for the values package is already to be able to manage values from a single point if there's need for it. This is a local scope and the variable declaration to its use is pretty close, so it's not necessary. Besides the former has shorter lines of code than the suggestion.

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+	item := func(t, v string) layout.Widget {
+		return func(gtx C) D {
+			l := pg.theme.Label(values.TextSize14, t)
+			r := pg.theme.Label(values.TextSize14, v)
```
```suggestion
			l := pg.theme.Body2(t)
			r := pg.theme.Body2(v)
```

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+
+	return card.Layout(gtx, func(gtx C) D {
+		m16 := values.MarginPadding16
+		return layout.Inset{Left: m16}.Layout(gtx, func(gtx C) D {
```
```suggestion
		return layout.Inset{Left: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
```

No need for the extra variable as it can fit into the Inset.

Sirmorrison commented on ui/statistics_page.go:
```diff
@@ -0,0 +1,120 @@
+			},
+			body: func(gtx C) D {
+				return pg.layoutStats(gtx)
+			},
```
```suggestion
			body: pg.layoutStats,
```

Sirmorrison reviewed (dismissed):
The uptime on the mockup is different from what we have implemented. From the mockup, it looks like the up time is a timer, to determine how long the app has been running.
<img width="591" alt="image" src="https://user-images.githubusercontent.com/27733432/120120296-309f5100-c194-11eb-88e7-b504004e5635.png">

Also, you currently have `testnet3` on the `Network` it should read either `testnet` or `mainnet`
<img width="573" alt="image" src="https://user-images.githubusercontent.com/27733432/120120698-e23f8180-c196-11eb-9804-b8daa19765f9.png">

Commit:
Update statistic page
- Use theme Separator in layout widget slice
- Correct uptime value
- Change network name with suffix to actual name
- Refresh frame for each 1 second
